connector/client/client.go

```diff
@@ -1,5 +1,122 @@
 package client
 
+import (
+	"bytes"
+	"encoding/json"
+	"net/http"
+	"net/url"
+	"strconv"
+	"strings"
+	"fmt"
+	"errors"
+
+	"github.com/infracloudio/msbotbuilder-go/connector/auth"
+	"github.com/infracloudio/msbotbuilder-go/schema"
+	"github.com/infracloudio/msbotbuilder-go/schema/customerror"
+)
+
 type Client interface {
 	Post(url url.URL, activity schema.Activity) error
 }
+
+type ConnectorClient struct {
+	Config ConnectorClientConfig
+}
+
+// Post a activity to given URL
+func (client ConnectorClient) Post(target url.URL, activity schema.Activity) error {
+
+	token, err := client.getToken()
+
+	if err != nil {
+		return err
+	}
+
+	jsonStr, err := json.Marshal(activity)
+
+	if err != nil {
+		return err
+	}
+
+	req, err := http.NewRequest("POST", target.String(), bytes.NewBuffer(jsonStr))
+
+	if err != nil {
+		return err
+	}
+
+	req.Header.Set("Content-Type", "application/json")
+	req.Header.Set("Authorization", "Bearer "+token)
+
+	replyClient := &http.Client{}
+	
+	resp, err := replyClient.Do(req)
+	
+
+	if err != nil || !(resp.StatusCode >= 200 && resp.StatusCode < 400) {
```

Comment:
Why do we need special handling on status code? Are we also treating non 200 status code as success? If so, can you please add a comment?

Reply:
Because the `Do` method doesn't always return non-nil `err`. One case I checked is for an incorrect URL, it returned `400` and non-nil `err`.
And since this is a supposed to be a generic client, I thought of generalizing the success codes as between 200 & 400.
The code that we expect for sending activity is specifically 201. So if we want to be specific, we can check 201 instead of 200-400

Reply:
Yes, that would be better.